bika/lims/adapters/widgetvisibility.py

```diff
@@ -230,3 +230,29 @@ def __call__(self, context, mode, field, default):
         if fieldName in fields and not ShowPrices:
             state = 'invisible'
         return state
+
+class SampleDateReceived(object):
+    """DateReceived is editable in sample context, only if all related analyses
+    are not yet submitted.
+    """
+    implements(IATWidgetVisibility)
+
+    def __init__(self, context):
+        self.context = context
+        self.sort = 3
+
+    def __call__(self, context, mode, field, default):
+        state = default if default else 'visible'
+        if field.getName() == 'DateReceived':
+            # if ONLY analyses in one of these states exist, then it is
+            # permitted to edit the field.
+            allowed_states = ['sample_received', 'retracted', 'rejected']
+            wf = get_tool('portal_workflow')
+            anwf = wf['bika_analysis_workflow']
+            state_ids = [sid for sid in anwf.states if sid not in allowed_states]
+            if context.getAnalyses(review_state=state_ids):
```

Comment:
If the context is AR, it will only look at the analysis the AR contains, but remember `DateReceived` refers to the Sample, that might contain more than one AR. Although we could get the Sample from the AR here, I think is better to keep things simple and only allow the edition of `DateReceived` in Sample view, not in AR view.

Reply:
Done